### PR TITLE
App Service RPC Tests

### PR DESCRIPTION
**Pull request by jacobkeeler**
ATF Test Scripts to check https://github.com/smartdevicelink/sdl_core/issues/2806

This PR is **not ready** for review.

### Summary
Adds success flow tests for PublishAppService and GetAppServiceData RPCs

### ATF version
[feature/app_services branch](https://github.com/smartdevicelink/sdl_atf/tree/feature/app_services)

##### Enhancements
* Adds success flow tests for PublishAppService and GetAppServiceData RPCs

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/AppServices/GetAppServiceData/001_M2M_success_flow.lua [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("RAI", common.registerApp)
+runner.Step("PTU", common.policyTableUpdate, { PTUfunc })
+runner.Step("RAI w/o PTU", common.registerAppWOPTU, { 2 })
+runner.Step("Activate App", common.activateApp)
+runner.Step("Publish App Service", common.publishMobileAppService, { manifest })
```
Comment (JackLivio) [inline test_scripts/AppServices/GetAppServiceData/001_M2M_success_flow.lua]: I am seeing a failure on this step

File: test_scripts/AppServices/GetAppServiceData/003_H2M_success_flow.lua [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("RAI", common.registerApp)
+runner.Step("PTU", common.policyTableUpdate, { PTUfunc })
+runner.Step("Activate App", common.activateApp)
+runner.Step("Publish App Service", common.publishMobileAppService, { manifest })
```
Comment (JackLivio) [inline test_scripts/AppServices/GetAppServiceData/003_H2M_success_flow.lua]: I am seeing a failure on this step

File: test_scripts/AppServices/commonAppServices.lua [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+function commonAppServices.publishMobileAppService(manifest, app_id)
+  if not app_id then app_id = 1 end
+  local mobileSession = commonAppServices.getMobileSession(app_id)
+  local cid = mobileSession:SendRPC("PublishAppService", {
+    appServiceManifest = manifest
+  })
+
+  local first_run = true
+  mobileSession:ExpectNotification("OnSystemCapabilityUpdated"):Times(AtLeast(1)):ValidIf(function(self, data)
```
Comment (JackLivio) [inline test_scripts/AppServices/commonAppServices.lua]: There should also be an expectation for the HMI to receive this notification.

File: test_scripts/AppServices/OnAppServiceData/004_OnAppServiceData_unknown_param_success.lua [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	--[[ Scenario ]]
+	runner.Title("Preconditions")
+	runner.Step("Clean environment", common.preconditions)
+	runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+	runner.Step("RAI", common.registerApp)
+	runner.Step("PTU", common.policyTableUpdate, { PTUfunc })
+	runner.Step("RAI w/o PTU", common.registerAppWOPTU, { 2 })
+	runner.Step("Activate App", common.activateApp)
+	runner.Step("Publish App Service", common.publishMobileAppService, { manifest })
+  runner.Step("Subscribe App Service Data", common.mobileSubscribeAppServiceData, { 1, 2 })
```
Comment (dboltovskyi) [inline test_scripts/AppServices/OnAppServiceData/004_OnAppServiceData_unknown_param_success.lua]: @jacobkeeler
Currently script fails on this step.
The reason is parameters for the `common.mobileSubscribeAppServiceData` function.
Seems these has to be changed to `{ 1, manifest.serviceType, 2 }`

File: test_sets/app_services.lua [hunk 5]
```diff
@@ -0,0 +1,57 @@
+./test_scripts/AppServices/OnAppServiceData/002_M2H_success_flow.lua
+./test_scripts/AppServices/OnAppServiceData/004_OnAppServiceData_unknown_param_success.lua
+./test_scripts/AppServices/OnAppServiceData/003_H2M_success_flow.lua
+./test_scripts/AppServices/OnAppServiceData/006_M2M_Weather_success_flow.lua
+./test_scripts/AppServices/OnAppServiceData/001_M2M_success_flow.lua
+./test_scripts/AppServices/OnAppServiceData/007_M2M_Future_service_type_success_flow.lua
+./test_scripts/AppServices/OnAppServiceData/005_M2M_Nav_success_flow.lua
+./test_scripts/AppServices/commonAppServices.lua
```
Comment (jacobkeeler) [inline test_sets/app_services.lua]: Delete this

File: test_sets/app_services.lua [hunk 6]
```diff
@@ -0,0 +1,57 @@
+./test_scripts/AppServices/OnAppServiceData/002_M2H_success_flow.lua
+./test_scripts/AppServices/OnAppServiceData/004_OnAppServiceData_unknown_param_success.lua
+./test_scripts/AppServices/OnAppServiceData/003_H2M_success_flow.lua
+./test_scripts/AppServices/OnAppServiceData/006_M2M_Weather_success_flow.lua
+./test_scripts/AppServices/OnAppServiceData/001_M2M_success_flow.lua
```
Comment (jacobkeeler) [inline test_sets/app_services.lua]: Minor detail, but these should probably be sorted by prefix. You could probably use a "Sort Lines" function with a text editor

File: test_scripts/AppServices/OnAppServiceData/004_OnAppServiceData_unknown_param_success.lua (hunk 4 above)
Comment (jacobkeeler) [reply test_scripts/AppServices/OnAppServiceData/004_OnAppServiceData_unknown_param_success.lua]: Fixed with https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2167/commits/80863e1eb5fb6594baa94a1c9af1118b69d89a2d

**Pull request merged**